Fixed cycle dependency

atsikham opened the pull request:
Related task is #6

sk4zuzu commented on resources/terraform/locals.tf:
```diff
@@ -1,4 +1,5 @@
 locals {
+  cluster_name                = "${var.name}-eks-cluster"
```
What about adding random string to all resources? :thinking: This is kinda good practice I believe.

sk4zuzu replied:
Terraform has "random" provider. :+1::innocent:

mkyc replied:
I think that this is good idea but only in "test" mode. I would prefer "test" boolean and if set to true, then add that random name. In "non-test" mode, I'd prefer those names to be predictable. I'd separate it to another task, to introduce test mode. Also in this test mode tag "test" might be added to all resources to easily do some "daily cleanup".

atsikham replied:
@mkyc Agree with you.

sk4zuzu replied:
So the random name would forever prevent some possible (but unlikely) name clashes. It's not that important, it's just what people do, we don't have to. For me personally the idea that something in the cloud is "not randomized" is counter-intuitive, but that's probably just me. :hugs:

mkyc replied:
My opinion is that if you create actual stable production environments you want to have something stable because eventually you want to point some DNS to some IP. I'm not saying that we don't want to do it, but I would not force randomisation. Tags are good as well, but I've seen multiple cases when names of resources were actually used in some automations. 

If we think about random part of names lets modify my previous comment and have parameter "randomise_names" instead of meaningless "test" or "test_mode". If you want to have names randomised you would just switch that parameter. 

This should go to another super low priority task.

sk4zuzu commented on resources/terraform/infra.tf:
```diff
@@ -18,11 +18,11 @@ resource "aws_subnet" "eks_subnet" {
-    name                                    = "${var.name}-eks-subnet${count.index}"
-    cluster_name                            = var.name
+    name                                          = "${var.name}-eks-subnet${count.index}"
+    cluster_name                                  = var.name
```
If cluster_name is name + something in locals then why cluster_name is just name here? :thinking:

mkyc reviewed (dismissed):
Consider cluster_name fix as suggested by @sk4zuzu

atsikham commented:
@mkyc How about names with random prefix but addition of constant `resource_group` tag? This will allow us to filter resources, but in the same time run tests in parallel for example.

atsikham commented:
Created new tasks to introduce test mode for all modules. Added tags to all possible resources to allow removal and filtering by them and removed cluster_name variable.

sk4zuzu reviewed (approved):
:monkey:

mkyc reviewed (approved):
LGTM
